src/Compilers/CSharp/Portable/Binder/Binder.WithQueryLambdaParametersBinder.cs

```diff
@@ -76,7 +76,7 @@ private BoundExpression SelectField(SimpleNameSyntax node, BoundExpression recei
                 {
                     // We only construct transparent query variables using anonymous types, so if we're trying to navigate through
                     // some other type, we must have some query API where the types don't match up as expected.
-                    var info = new CSDiagnosticInfo(ErrorCode.ERR_UnsupportedTransparentIdentifierAccess, name, receiver.ExpressionSymbol ?? receiverType);
+                    var info = new CSDiagnosticInfo(ErrorCode.ERR_UnsupportedTransparentIdentifierAccess, name, new FormattedSymbol(receiver.ExpressionSymbol ?? receiverType, SymbolDisplayFormat.CSharpErrorMessageTypeFormat));
```

Comment:
>new FormattedSymbol(receiver.ExpressionSymbol ?? receiverType, SymbolDisplayFormat.CSharpErrorMessageTypeFormat)) [](http://example.com/codeflow?start=112&length=113)

It is not clear why would we want to use ```CSharpErrorMessageTypeFormat``` for ```receiver.ExpressionSymbol```. Is there a guarantee that ```receiver.ExpressionSymbol``` is always a type here?

Also, I am curious why do we need to display types specially here.

Reply:
> Is there a guarantee that `receiver.ExpressionSymbol` is always a type here?

No, in the only test that covers this diagnostic ([`QueryTests.BrokenQueryPattern`](https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.Semantic.UnitTests/Semantics/QueryTests.cs,709728f03d48b4bc)), the receiver is actually a `ParameterSymbol`, but it has compiler-generated name `int <>h__TransparentIdentifier0` (which should be always the case for this diagnostic if I understand it correctly). That's why I kept the existing behavior of displaying just the type.